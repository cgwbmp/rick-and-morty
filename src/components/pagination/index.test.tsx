import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Pagination from '.';

describe('<Pagination />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Pagination page={1} pages={2} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('DOM structure not changed', () => {
    const component = renderer.create(<Pagination page={1} pages={2} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
