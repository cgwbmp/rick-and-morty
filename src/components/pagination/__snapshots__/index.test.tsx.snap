// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> DOM structure not changed 1`] = `
<div
  className="pagination"
>
  <button
    className="button"
    disabled={true}
    type="button"
  >
    Previos page
  </button>
  <span
    className="status"
  >
    1/2
  </span>
  <a
    className="link"
    href="/?page=2"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Next page
  </a>
</div>
`;
